Collision/src/main/java/dk/gruppeseks/bodtrd/collision/CollisionProcessor.java

```diff
@@ -96,7 +102,7 @@ else if (ent.getType() == EntityType.WALL)
             }
             case ENEMY:
             {
-                if (ent.getType() == EntityType.WALL)
+                if (ent.get(CollisionData.class).getCollisionFlag() == CollisionFlags.CONCRETE)
```

Comment:
the correct way to check this is `if ((ent.get(CollisionData.class).getCollisionFlag() & CollisionFlags.CONCRETE) != 0)`